Block/Widget/ExpressCheckoutButton.php

```diff
@@ -51,16 +59,16 @@ public function __construct(
         Data                        $urlHelper, array $data = []
     )
     {
-        $this->_productFactory = $productFactory;
+        $this->productFactory = $productFactory;
         $this->productRepository = $productRepository;
         $this->random = $random;
         $this->config = $config;
         $this->storeManager = $storeManager;
-        parent::__construct($context, $postDataHelper, $layerResolver, $categoryRepository, $urlHelper, $data);
         $this->setTemplate("Amwal_Payments::express/widget-checkout-button.phtml");
+        parent::__construct($context, $postDataHelper, $layerResolver, $categoryRepository, $urlHelper, $data);
```

Comment:
We should make sure that setTemplate is not called in the parent constructor, else our call is immediately overwritten